packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceRangeMapAndExporter.cpp

```diff
@@ -0,0 +1,318 @@
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+*/
+
+#include "Tpetra_TestingUtilities.hpp"
+#include "Tpetra_MultiVector.hpp"
+#include "Tpetra_CrsMatrix.hpp"
+
+namespace {
+
+  using std::endl;
+  using std::string;
+
+  using Teuchos::as;
+  using Teuchos::FancyOStream;
+  using Teuchos::RCP;
+  using Teuchos::ArrayRCP;
+  using Teuchos::rcp;
+  using Teuchos::arcp;
+  using Teuchos::outArg;
+  using Teuchos::arcpClone;
+  using Teuchos::arrayView;
+  using Teuchos::broadcast;
+  using Teuchos::OrdinalTraits;
+  using Teuchos::ScalarTraits;
+  using Teuchos::Comm;
+  using Teuchos::Array;
+  using Teuchos::ArrayView;
+  using Teuchos::tuple;
+  using Teuchos::null;
+  using Teuchos::ParameterList;
+  using Teuchos::parameterList;
+
+  using Tpetra::Map;
+  using Tpetra::MultiVector;
+  using Tpetra::Vector;
+  using Tpetra::Operator;
+  using Tpetra::CrsMatrix;
+  using Tpetra::CrsGraph;
+  using Tpetra::RowMatrix;
+  using Tpetra::Export;
+  using Tpetra::global_size_t;
+  using Tpetra::createContigMapWithNode;
+  using Tpetra::createVector;
+  using Tpetra::ProfileType;
+  using Tpetra::StaticProfile;
+  using Tpetra::OptimizeOption;
+  using Tpetra::DoOptimizeStorage;
+  using Tpetra::DoNotOptimizeStorage;
+  using Tpetra::GloballyDistributed;
+  using Tpetra::INSERT;
+
+  TEUCHOS_STATIC_SETUP()
+  {
+    Teuchos::CommandLineProcessor &clp = Teuchos::UnitTestRepository::getCLP();
+    clp.addOutputSetupOptions(true);
+  }
+
+  //
+  // UNIT TESTS
+  //
+#define applyAndCheckResult( A, B, newMap) \
```

Comment:
same comment on this function as in the previous file.  Also, if it's repeated across separate files, shouldn't it go in a header to reduce duplication?

Reply:
Same solution as above.
The functions are different; one imports before the apply, the other exports after the apply.

Reply:
I see, thanks